File: recursive.exs
defmodule Factorial do
  def factorial(n) do
    if n < 1 do
      1
    else
      n * factorial(n-1)
    end
  end
end

input_ = String.trim(IO.gets("Number: "))
{input, ""} = Integer.parse(input_)

IO.puts Factorial.factorial(input) 

File: recursive.go
package main

import "fmt"

func factorial(n int) int {
  if n < 1 {
    return 1
  } else {
    return n * factorial(n - 1)
  }
}

func main() {
  var input int
  fmt.Print("Number: ")
  fmt.Scanln(&input)
  fmt.Println(factorial(input))
}
